### ISSUE TITLE
Do not suggest to remove generic names from landuse and other mismatching elements

### ISSUE THREAD

**Issue by matkoniecz** [openstreetmap/iD#5949]
Some people (especially beginners) use landuse instead of shop tags.

For example https://www.openstreetmap.org/way/435769787 was tagged as `landuse=retail name=bakery`

iD in that situation should not propose removal of `name=bakery`

This name should be removed as generic solely from `shop=bakery`.

I propose to stop suggesting of generic names from landuses - and maybe propose switching to correct object type (preset for bakery may have list of generic names used for it).

![selection_001](https://user-images.githubusercontent.com/899988/53297918-e9dc8700-3825-11e9-94ae-942f4140e0d5.png)

`landuse=retail` without name because it was removed is much harder to fix than `landuse=retail name=bakery` that is really easy to fix, so removing generic names from generic landuses is harmful.

Proposing to switch directly to `shop=bakery` would be probably even better.

**Comment by matkoniecz** [comment]
Other example = `shop=yes` + `name=Bakery` (or like in my case `name=Piekarnia`).

**Comment by BjornRasmussen** [comment]
This would require a list of what generic names should be had as.  Perhaps a list of generic names could be added to each preset rather than the names being stored in a single list?

**Comment by DujaOSM** [comment]
Apologies for hijacking this semi-related issue, but what happened in Update 2.15.0 (#2231)? It seems that generic names warning is removed altogether? What does the "Suspicious Names" validation exactly do now?

I was using this feature a lot, assigning temporary labels to otherwise unnamed landuse relations (Farmland "1", Scrub "6") solely for editing purposes (much easier to build relations this way), and I was warned before commit, so I could remove those names. Now there's no warning at all, so I tend to leave garbage around...

**Comment by matkoniecz** [comment]
Apologies for hijacking this semi-related issue, but what happened in Update 2.15.0? It seems that generic names warning is removed altogether?

It seems to be present, I just tested with `name=hotel`

> Line has the suspicious name "hotel"

**Comment by bhousel** [comment]
> Apologies for hijacking this semi-related issue, but what happened in Update 2.15.0? It seems that generic names warning is removed altogether? What does the "Suspicious Names" validation exactly do now?

The validation is still there, it's just that the source list has changed to not include names like the temporary ones you were using.  We don't want to warn people about "Motel 6" after all.

**Comment by DujaOSM** [comment]
> It seems to be present, I just tested with `name=hotel`
Well, I tested it with a relation called "farmland", a cemetery called "cemetery", and an orchard called "orchard" and it did not complain. So it seems to be of very limited use.
Before the number-as-name validation was introduced, I used special characters such as `. , _ / :` in order to minimize the impact of potential commit, but the iD warning was great. 
Can I get my toy back in some form, please? I can't think of many situations when a number would constitute a valid name for anything, and non-alphanumeric characters in particular.

**Comment by DujaOSM** [comment]
> Well, I tested it with a relation called "farmland", a cemetery called "cemetery", and an orchard called "orchard" and it did not complain. 
But on the other hand, it did issue a warning
> Farmland has the suspicious name "hotel"
so I have my workaround. :)

**Comment by bhousel** [comment]
TBH I don't know how this issue is still open.  If you add the name "bakery" to a landuse, it's totally ok for iD to warn about it.

**Comment by matkoniecz** [comment]
> it's totally ok for iD to warn about it

The problem is that iD is not just warning, it is encouraging mapper to remove it

For example https://www.openstreetmap.org/way/435769787 was tagged as `landuse=retail name=bakery`

iD in that situation should not propose removal of `name=bakery`, it is not helpful - the proper fix in that case was to move it to `shop=bakery`

**Comment by matkoniecz** [comment]
Note that title is

> Do not suggest to remove generic names from landuse

not

> Do not warn about generic names on landuse

**Comment by 1ec5** [comment]
Perhaps a solution would be to add a second option below it, for removing the landuse tag. Choosing that option would trigger the error about missing a feature tag, which has its own suggestion to choose an appropriate feature tag. That way, we don’t need to fuzzy-match on preset names, which could get messy.

**Comment by bhousel** [comment]
> iD in that situation should not propose removal of `name=bakery`, it is not helpful - the proper fix in that case was to move it to `shop=bakery`

Got it - I was indeed confused by the title.

**Comment by bhousel** [comment]
> In any case, I don't see this fix as critical. It would be nice to have, but how often does a feature's preset mismatch with its name anyway?

Yeah we've gone back and forth a few times on this issue, but I think what they are really asking for is to check if `name=thing` happens to match a preset for `shop=thing` (or `amenity=thing` etc),  and if so to offer to switch to that other preset and remove the name.  That's pretty easy I think.

For generic names, iD currently suggests to "remove the name" which is kind of not the correct fix, but at least calls attention to the issue.

**Comment by 1ec5** [comment]
> check if `name=thìng happens to match a preset for `shop=thìng (or `amenity=thing` etc)

Seems like users would be less likely to tag `name=thìng with the main workflow in v3, which forces the user to choose a preset before they’d have a chance to name the feature.

**Comment by quincylvania** [comment]
> For generic names, iD currently suggests to "remove the name" which is kind of not the correct fix, but at least calls attention to the issue.

It's true this doesn't fit in all cases, but it's useful when a feature is correctly tagged yet has a redundant, generic name. Like `amenity=drinking_water` with `name=water fountain`.

**Comment by matkoniecz** [comment]
thinking about this again: rules are fiddly enough, problem is rare enough that random issue from https://github.com/openstreetmap/iD/issues is likely more worthwhile to implement
